src/rp2_common/pico_runtime_init/runtime_init_clocks.c

```diff
@@ -94,6 +87,15 @@ void __weak runtime_init_clocks(void) {
                         0,
                         XOSC_HZ);
 
+        // This must be done after we've configured CLK_REF to XOSC due to the need to time a delay
+#if SYS_CLK_VREG_VOLTAGE_AUTO_ADJUST && defined(SYS_CLK_VREG_VOLTAGE_MIN)
+        if (vreg_get_voltage() < SYS_CLK_VREG_VOLTAGE_MIN) {
+            vreg_set_voltage(SYS_CLK_VREG_VOLTAGE_MIN);
+            // wait for voltage to settle; must use CPU cycles as TIMER is not yet clocked correctly
+            busy_wait_at_least_cycles((uint32_t)((SYS_CLK_VREG_VOLTAGE_AUTO_ADJUST_DELAY_US * (uint64_t)XOSC_HZ) / 1000000));
```

Comment:
This might be a stupid question, but given that there's a division by 1000000 in there, would it be possible to rearrange this calculation to not use 64-bit arithmetic? (given that we've already got at least one PR that is attempting to reduce the bit of 64-bit maths)

Reply:
it is a compile time constant, so handled by the compiler